extensions/azure-extensions/src/main/java/io/druid/storage/azure/AzureStorageContainer.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Druid - a distributed column store.
+ *  Copyright 2012 - 2015 Metamarkets Group Inc.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package io.druid.storage.azure;
+
+import com.metamx.common.logger.Logger;
+import com.microsoft.azure.storage.StorageException;
+import com.microsoft.azure.storage.blob.CloudBlob;
+import com.microsoft.azure.storage.blob.CloudBlobContainer;
+import com.microsoft.azure.storage.blob.CloudBlobDirectory;
+import com.microsoft.azure.storage.blob.CloudBlockBlob;
+import com.microsoft.azure.storage.blob.ListBlobItem;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URISyntaxException;
+import java.util.ArrayList;
+import java.util.List;
+
+public class AzureStorageContainer
+{
+
+  private final Logger log = new Logger(AzureStorageContainer.class);
+
+  private final CloudBlobContainer cloudBlobContainer;
+
+  public AzureStorageContainer(
+      CloudBlobContainer cloudBlobContainer
+  )
+  {
+    this.cloudBlobContainer = cloudBlobContainer;
+  }
+
+  public List<String> emptyCloudBlobDirectory(final String path)
+      throws StorageException, URISyntaxException
+  {
+    List<String> deletedFiles = new ArrayList<>();
+
+    CloudBlobDirectory cloudBlobDirectory = cloudBlobContainer.getDirectoryReference(path);
+    for (ListBlobItem blobItem : cloudBlobDirectory.listBlobs()) {
+      if (blobItem instanceof CloudBlob) {
```

Comment:
The code was written in that way because the files will be located under the given path directory and `cloudBlobDirectory.listBlobs()` can return files (`CloudBlob`) or virtual directories (`CloudBlobDirectory`).

I'll change the code to get a flat list of all files under the given path and then delete each one of them. If no file was deleted, a warn log message will be filed.